modal/_container_io_manager.py

```diff
@@ -101,8 +102,9 @@ def _init(self, container_args: api_pb2.ContainerArguments, client: _Client):
 
         self._semaphore = None
         self._environment_name = container_args.environment_name
-        self._waiting_for_memory_snapshot = False
         self._heartbeat_loop = None
+        self._pause_heartbeats = asyncio.Condition()
```

Comment:
Would prefer a different name here. "pause heartbeats" is an imperative mood, which is typical of functions. Something like `self._heartbeat_condition` is fine to me.